### ISSUE TITLE
Twitch stream live indicator in Usercard

### ISSUE THREAD

**Issue by auxtinn** [Chatterino/chatterino2#2699]
# **What should be added?** #

**A LIVE or OFFLINE indicator should be shown (respectively) in a user's usercard to allow Chatterino users to easily identify whether or not the person is LIVE or OFFLINE.**
_**If you need more clarification on this, let me know in the comments.**_

# **Why should it be added?** #

**I believe this should be added so that users of Chatterino can know whether someone is live or not by just clicking on their username or whatnot to bring up the person's usercard and see some type of LIVE or OFFLINE indicator there.**

**Comment by Mm2PL** [comment]
How would this look in your opinion?

**Comment by AustinHolderr** [comment]
> How would this look in your opinion?

not quite sure, just an idea that sparked and came across my mind, so i thought i would open up an 'issue' here. it could be anything from a dot like what is shows on tabs that are open in Chatterino when a channel is live, or it could be a red dot (to the left of the text) then a red text indicator saying LIVE and for OFFLINE just be basically the same thing like have a grey dot (to the left of the text) and then say OFFLINE in a grey text. just my thoughts right now...

**Comment by AustinHolderr** [comment]
> > How would this look in your opinion?
> 
> not quite sure, just an idea that sparked and came across my mind, so i thought i would open up an 'issue' here. it could be anything from a dot like what is shows on tabs that are open in Chatterino when a channel is live, or it could be a red dot (to the left of the text) then a red text indicator saying LIVE and for OFFLINE just be basically the same thing like have a grey dot (to the left of the text) and then say OFFLINE in a grey text. just my thoughts right now...

# **somewhere in and around right here...** #
<img width="438" alt="Chatterino Usercard LIVE/OFFLINE Indicator(s)" src="https://user-images.githubusercontent.com/44728118/116158027-f71e8600-a6bb-11eb-990e-61692387ff39.png">

**Comment by jammehcow** [comment]
Would it not make sense to just outline the user's profile picture with a colour (maybe the channel's custom colour if that's doable) since clicking the profile picture takes you to the Twitch channel anyways?

A badly designed example:
![116158027-f71e8600-a6bb-11eb-990e-61692387ff39](https://user-images.githubusercontent.com/4962764/116167404-3dbfbf00-a754-11eb-89d6-f8bb6efca249.png)

**Comment by alazymeme** [comment]
I have two options to suggest:

- Red circle next to the user's username
- Under the ID, but say something like `Live for <uptime>` instead of just stating that the user is live

**Comment by apa420** [comment]
I have a really stupid question, but why?
I don't see why you need to see if a random person in chat is live or not pressing his usercard.

**Comment by zneix** [comment]
I can agree here, I think it's a very niche thing to be present in usercard and we don't want to add unnecessary clutter. However, this is just my personal opinion.
Worth noting that twitch also doesn't tell you whether a user is live or not in their usercard.
![this is quite useless](https://cdn.zneix.eu/qKwidNQ.png)

**Comment by AustinHolderr** [comment]
> Would it not make sense to just outline the user's profile picture with a colour (maybe the channel's custom colour if that's doable) since clicking the profile picture takes you to the Twitch channel anyways?
> 
> A badly designed example:
> ![116158027-f71e8600-a6bb-11eb-990e-61692387ff39](https://user-images.githubusercontent.com/4962764/116167404-3dbfbf00-a754-11eb-89d6-f8bb6efca249.png)
> 
> EDIT: also would be nice to show a tooltip to the right of the image (statically positioned) which says 'live'

could do that too

**Comment by AustinHolderr** [comment]
> I have two options to suggest:
> 
> * Red circle next to the user's username
> * Under the ID, but say something like `Live for <uptime>` instead of just stating that the user is live

i like the uptime idea

**Comment by AustinHolderr** [comment]
> I have a really stupid question, but why?
> I don't see why you need to see if a random person in chat is live or not pressing his usercard.

it would just be something nice to have.

**Comment by AustinHolderr** [comment]
> I can agree here, I think it's a very niche thing to be present in usercard and we don't want to add unnecessary clutter. However, this is just my personal opinion.
> Worth noting that twitch also doesn't tell you whether a user is live or not in their usercard.
> ![this is quite useless](https://camo.githubusercontent.com/1f6a2d4b2a59126435eba1b3932ae0a3523a309a789db621f4aa2dbdd118f8b0/68747470733a2f2f63646e2e7a6e6569782e65752f714b7769644e512e706e67)

true, but i do feel like it would be a nice feature to have added to chatterino

**Comment by apa420** [comment]
> # **Why should it be added?**
> 
> **I believe this should be added so that users of Chatterino can know whether someone is live or not by just clicking on their username or whatnot to bring up the person's usercard and see some type of LIVE or OFFLINE indicator there.**

This does not answer the question it's just the idea written differently. I still don't understand the purpose of doing an api call for this.

**Comment by AustinHolderr** [comment]
> > # **Why should it be added?**
> 
> > 
> 
> > **I believe this should be added so that users of Chatterino can know whether someone is live or not by just clicking on their username or whatnot to bring up the person's usercard and see some type of LIVE or OFFLINE indicator there.**
> 
> 
> 
> This does not answer the question it's just the idea written differently. I still don't understand the purpose of doing an api call for this.

well everyone else understood it..

**Comment by sando** [comment]
> > > # **Why should it be added?**
> > 
> > 
> > > 
> > 
> > 
> > > **I believe this should be added so that users of Chatterino can know whether someone is live or not by just clicking on their username or whatnot to bring up the person's usercard and see some type of LIVE or OFFLINE indicator there.**
> > 
> > 
> > This does not answer the question it's just the idea written differently. I still don't understand the purpose of doing an api call for this.
> 
> well everyone else understood it..

What I assume @apa420 wanted clarification on is *why should this functionality be added*, i.e. what is the use case?

To reword that question, what is the purpose of spending another API call on seeing if a user is online or not from the context of opening their usercard? I'm not sure if I see the point either.

**Comment by AustinHolderr** [comment]
> > > > # **Why should it be added?**
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > **I believe this should be added so that users of Chatterino can know whether someone is live or not by just clicking on their username or whatnot to bring up the person's usercard and see some type of LIVE or OFFLINE indicator there.**
> 
> > > 
> 
> > > 
> 
> > > This does not answer the question it's just the idea written differently. I still don't understand the purpose of doing an api call for this.
> 
> > 
> 
> > well everyone else understood it..
> 
> 
> 
> What I assume @apa420 wanted clarification on is *why should this functionality be added*, i.e. what is the use case?
> 
> 
> 
> To reword that question, what is the purpose of spending another API call on seeing if a user is live or not from the context of opening their usercard? I'm not sure if I see the point of knowing if they're live either.

some community members like to support each others' streams and if you don’t know they're live, you can’t support their stream by lurking and whatnot. it's helpful because you can then easily tell while they're lurking in the first stream where everyone is, to be able to know just by clicking on their name if they're live or not instead of having to open up a whole new tab in chatterino for it. sorry if this didn't make a lot of sense, it's late at night and i’m a bit tired from the day...

**Comment by TranRed** [comment]
> some community members like to support each others' streams and if you don’t know they're live, you can’t support their stream by lurking and whatnot. it's helpful because you can then easily tell while they're lurking in the first stream where everyone is, to be able to know just by clicking on their name if they're live or not instead of having to open up a whole new tab in chatterino for it. sorry if this didn't make a lot of sense, it's late at night and i’m a bit tired from the day...

Can this usecase be covered by using the /live tab?
